apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -544,42 +651,200 @@ maps_contract(Config) ->
-    ContractString = aeso_test_utils:read_contract("maps"),
-    BinCode = aeso_compiler:from_string(ContractString, []),
-    HexCode = aeu_hex:hexstring_encode(BinCode),
+    %% Compile test contract "maps.aes".
+    MCode = compile_test_contract("maps"),
 
     %% Initialise contract owned by Alice.
-    {EncodedContractPubkey,_,_} =
-	create_contract(NodeName, APubkey, APrivkey, HexCode, <<"()">>),
+    {EncodedMapsPubkey,DecodedMapsPubkey,_} =
+       create_compute_contract(NodeName, APubkey, APrivkey, MCode, <<"()">>),
 
-    %% get_state
-    ct:pal("Calling get_state\n"),
-    GetState1 = call_get_state(NodeName, BPubkey, BPrivkey,
-			       EncodedContractPubkey, HexCode),
-    ct:pal("Get state 1 ~p\n", [GetState1]),
+    %% Some functions can't be called directly, we need an interface contract.
+    TestMapsFile = proplists:get_value(data_dir, Config) ++ "test_maps.aes",
+    {ok,SophiaCode} = file:read_file(TestMapsFile),
+    {ok, 200, #{<<"bytecode">> := TCode}} = get_contract_bytecode(SophiaCode),
 
-    %% Set state {[k] = v}
-    call_func(NodeName, BPubkey, BPrivkey, EncodedContractPubkey, HexCode,
-			      <<"map_state_i">>, <<"()">>),
-    call_func(NodeName, CPubkey, CPrivkey, EncodedContractPubkey, HexCode,
-			      <<"map_state_s">>, <<"()">>),
+    {EncodedTestPubkey,_,_} =
+        create_compute_contract(NodeName, APubkey, APrivkey, TCode,
+                                args_to_binary([DecodedMapsPubkey])),
 
-    %% get_state
-    ct:pal("Calling get_state\n"),
-    GetState2 = call_get_state(NodeName, APubkey, APrivkey,
-			       EncodedContractPubkey, HexCode),
-    ct:pal("Get state 2 ~p\n", [GetState2]),
+    %% Set state {[k] = v}
+    %% State now {map_i = {[1]=>{x=1,y=2},[2]=>{x=3,y=4},[3]=>{x=5,y=6}},
+    %%            map_s = ["one"]=> ... , ["two"]=> ... , ["three"] => ...}
+    call_compute_func(NodeName, BPubkey, BPrivkey, EncodedMapsPubkey,
+                      <<"map_state_i">>, <<"()">>),
+    call_compute_func(NodeName, CPubkey, CPrivkey, EncodedMapsPubkey,
+                      <<"map_state_s">>, <<"()">>),
+
+    %% Print current state
+    ct:pal("State ~p\n", [call_get_state(NodeName, APubkey, APrivkey,
+                                         EncodedMapsPubkey)]),
+
+    %% m[k]
+
+    [#{<<"type">> := <<"word">>, <<"value">> := 3},
+     #{<<"type">> := <<"word">>, <<"value">> := 4}] =
+        call_get_state_i(NodeName, BPubkey, BPrivkey, EncodedMapsPubkey,
+                         <<"(2)">>),
+    [#{<<"type">> := <<"word">>, <<"value">> := 5},
+     #{<<"type">> := <<"word">>, <<"value">> := 6}] =
+        call_get_state_s(NodeName, BPubkey, BPrivkey, EncodedMapsPubkey,
+                         <<"(\"three\")">>),
+
+    %% m{[k] = v}
+    %% State now {map_i = {[1]=>{x=11,y=22},[2]=>{x=3,y=4},[3]=>{x=5,y=6}},
+    %%            map_s = ["one"]=> ... , ["two"]=> ... , ["three"] => ...}
+    %% Need to call interface functions as cannot create record as argument.
```

Comment:
I see that the [memory layout of Sophia encodes a record as the tuple of values](https://github.com/aeternity/protocol/blame/f8b99f42ea894d059a58ec4ebee2ce976415b6d8/contracts/sophia.md#L880): shall a contract call transaction have as input simply the tuple of the record values (in order)?

Reply:
I don't think so, hopefully not. Haven't tried actually. This does it cleanly and is type correct.

Reply:
Especially now that you need to define the types of functions you call in other contracts.

Reply:
Opened PR aimed at clarifying this https://github.com/aeternity/protocol/pull/198